### DISCUSSION TITLE
How to get ImageMagick to work in Razor .NET Core Website

### DISCUSSION THREAD

**Discussion by GitHubSlob** [dlemstra/Magick.NET#1524] [Help]
ImageMagick looks like something that would work for me.

The code below uploads a file. I want to upload an image and resize it to a "fixed size maintain the ratio".
I have installed the NuGet package called Magick.NET Core in VS 2019 but have no idea on how to reference it.
I would love to have ImageMagick work.  I would pay for the license but I have no idea and the samples just reference a local file but I want to manipulate and save the user's image that they upload.

Here is my simple file upload page that works and stores an image in a folder on my dev server.

This is barebones for illustration minus any future validation and database updating. 

But anything with the word "ImageMagik" in it is underlined in red. I have "using ImageMagick;" at the top of the page.

I will gladly share my completed Image upload/manipulate code with this group.

```C#
using System.IO;
using System.Threading.Tasks;
using Microsoft.AspNetCore.Http;
using Microsoft.AspNetCore.Mvc;
using Microsoft.AspNetCore.Mvc.RazorPages;
using ImageMagick;

namespace DaBamba.Pages.YourDances
{
    public class UploadFileModel : PageModel
    {
        private readonly Microsoft.AspNetCore.Hosting.IWebHostEnvironment _environment;
        public UploadFileModel(Microsoft.AspNetCore.Hosting.IWebHostEnvironment environment)
        {
            _environment = environment;
        }
        [BindProperty]
        public IFormFile Upload { get; set; }
        public async Task OnPostAsync()
        {

            var file = "C:\\inetpub\\wwwroot\\DaBamba\\wwwroot\\images\\DanceImages\\" + Upload.FileName;

            using var fileStream = new FileStream(file, FileMode.Create);

            //RESIZE IMAGE HERE

           await Upload.CopyToAsync(fileStream);

        }
    }
}
```

**Comment by dlemstra** [comment]
Which NuGet package did you install? It looks like you only installed the `Magick.NET.Core` library? You can find more information in the docs about which version you should install. But feel free to ask more questions here if you have any.

**Comment by GitHubSlob** [comment]
Hi Dirk,
I have installed Magick.NET.Core (13.5.0)
but I could install any package that is on NuGet.

Sincerely,
John A Davis

**Comment by dlemstra** [comment]
That .Core library is a dependency of the packages listed in the readme of this project. You will need to install one of those instead. To decide which one I would advise you to read the docs.

**Comment by GitHubSlob** [comment]
Ok, thanks for the clarification.
I am still going to try to figure this out.
I have been flat on my back with a fractured pelvis for the last 6 weeks.
Heading off to read the docs on GitHub

Thank you very much.
John A Davis

**Reply by dlemstra** [reply]
Just replying here so you know that your message has also been posted on GitHub when you responded to the email.
